apps/language_server/lib/language_server/providers/workspace_symbols.ex

```diff
@@ -0,0 +1,468 @@
+defmodule ElixirLS.LanguageServer.Providers.WorkspaceSymbols do
```

Comment:
In my tests (in vscode) it works as expected. When a project is loaded, the build and dialyzer starts automatically. The same happens if I switch branches (unless there is a difference in deps which simply crashes elixir-ls with `** (Mix.Error) Can't continue due to errors on dependencies`)
There is a quirk though. WorkspaceSymbols now depends on ElixirLS.LanguageServer.Dialyzer.SuccessTypings as this module loads all modules from all applications in the project and std lib basing on dialyzer PLTs. I tried launching WorkspaceSymbols indexing proces earlier (on start or just after the build) but isn't working correctly in the current architecture - the modules are simply not yet loaded.
Regarding the index being out of date - this happens a lot to dialyzer PLTs as well. I often end up with outdated dialyzer warnings that won't go away until restart.

Reply:
Ah you are correct. I think perhaps I didn't wait long enough for the initial compile to complete, ideally we could have `workspace/symbol` not finish responding until the first compile is complete, but if that's too much complexity we can look into adding it later.

> There is a quirk though. WorkspaceSymbols now depends on ElixirLS.LanguageServer.Dialyzer.SuccessTypings as this module loads all modules from all applications in the project and std lib basing on dialyzer PLTs

Hmmm, this means that if the user disables dialyzer in their settings that they cannot get workspace symbol support. I think for now we can just add that to list of known issues in the readme. Then later perhaps we could re-architect so that workspace symbols and the dialyzer task use shared code to load the list of modules (so it will still work if the dialyzer is disabled).

Reply:
Added known issue in https://github.com/elixir-lsp/elixir-ls/pull/110/commits/2e71ddb831291bb63f42f24bc8bf2fcae9ea4e00